tensorflow-framework/src/main/java/org/tensorflow/framework/metrics/impl/MetricsHelper.java

```diff
@@ -85,7 +92,8 @@
       }
 
       for (int i = 0; i < valuesRankStatic; i++) {
-        if (valuesShapeStatic.size(i) != weightsShapeStatic.size(i)) {
+        if (valuesShapeStatic.size(i) != weightsShapeStatic.size(i)
```

Comment:
I am not quite sure if it can happen. Whenever I try to create a shape with -1 in a dim, the system seems to fill this in when I fill the object with data. This is true with numpy and TF Java. At any rate to be safe, how about?

Change
`if (weightsRankStatic != Shape.UNKNOWN_SIZE && valuesRankStatic != Shape.UNKNOWN_SIZE) {`

to 
```
if(!weightsShapeStatic.isUnknown() && !valuesShapeStatic.isUnknown() &&
        !weightsShapeStatic.hasUnknownDimension() & !valuesShapeStatic.hasUnknownDimension()) {
```

If it fails this, then it falls into dynamic checks

Reply:
👍 , except that the final `&` -> `&&`